src/plugins/intel_cpu/src/nodes/stft.cpp

```diff
@@ -73,15 +77,101 @@ bool STFT::created() const {
+                      const float* window,
+                      float* rdft_result,
+                      const VectorDims& signal_shape,
+                      const VectorDims& window_shape,
+                      const int64_t frame_size,
+                      const int64_t frame_step,
+                      const bool transpose_frames,
+                      std::shared_ptr<RDFTExecutor> rdft_executor) {
+    const auto is_signal_1D = signal_shape.size() == 1;
+    const size_t batch_size = is_signal_1D ? 1 : signal_shape[0];
+    const size_t signal_axis = is_signal_1D ? 0 : 1;
+    const auto signal_length = signal_shape[signal_axis];
+    const auto num_frames = static_cast<size_t>((signal_length - frame_size) / frame_step) + 1;
+    const auto frame_size_dim = static_cast<size_t>(frame_size);
+    const auto fft_out_shape = VectorDims{static_cast<size_t>((frame_size_dim / 2) + 1), 2};
+    const auto fft_out_shape_size = shape_size(fft_out_shape);
+
+    const auto window_length = window_shape[0] < frame_size_dim ? window_shape[0] : frame_size_dim;
+    std::vector<float> pad_window(frame_size, 0);
+    cpu_parallel_memcpy(pad_window.data() + (frame_size_dim - window_length) / 2,
+                        window,
+                        sizeof(float) * window_shape[0]);
+
+    parallel_for2d(batch_size, num_frames, [&](size_t batch, size_t frame_idx) {
+        size_t batch_in_start = batch * signal_length;
+        size_t batch_frames_out = batch * num_frames;
+
+        const auto frame_start = batch_in_start + frame_idx * frame_step;
+        const auto frame_end = frame_start + frame_size;
+        std::vector<float> signal_slice(signal + frame_start, signal + frame_end);
+        std::transform(signal_slice.begin(),
+                       signal_slice.end(),
+                       pad_window.begin(),
+                       signal_slice.begin(),
+                       std::multiplies<float>());
+
+        const auto result_idx = (batch_frames_out + frame_idx) * fft_out_shape_size;
+        auto twiddles = rdft_executor->generateTwiddles({static_cast<int>(signal_slice.size())}, fft_out_shape, {0});
+        rdft_executor->execute(signal_slice.data(),
+                               rdft_result + result_idx,
+                               twiddles,
+                               1,
+                               {0},
+                               {static_cast<int>(frame_size)},
+                               {frame_size_dim},
+                               fft_out_shape,
+                               {1},
+                               {2, 1});
+    });
+    if (transpose_frames) {
+        const auto stft_shape = VectorDims{batch_size, num_frames, fft_out_shape[0], fft_out_shape[1]};
+        const auto stft_transp_out_shape = VectorDims{batch_size, fft_out_shape[0], num_frames, fft_out_shape[1]};
+        std::vector<float> signal_t(rdft_result, rdft_result + shape_size(stft_transp_out_shape));
```

Comment:
For the memory consumption optimization:
If `rdft_result` might be quite big blob, we can use plugin memory scratchpad here instead - this memory is shared between nodes and can be reused.
```
MemoryPtr dst_mem = nullptr;
float* dst = rdft_result;
if (transpose_frames) {  // we need to store intermediate results
   dst_mem = getScratchPadMem(std::make_shared<CpuBlockedMemoryDesc>(ov::element::u8, stft_transp_out_shape));  // Moreover, it might be moved to `PrepareParams`
   dst = dst_mem->getDataAs<uint8_t>();
}
<--- parallel_for2d --->
if (transpose_frames) {
   const auto stft_shape = VectorDims{batch_size, num_frames, fft_out_shape[0], fft_out_shape[1]};
   const auto stft_transp_out_shape = VectorDims{batch_size, fft_out_shape[0], num_frames, fft_out_shape[1]};
   transpose_out4d(reinterpret_cast<const uint8_t*>(dst),
                               reinterpret_cast<uint8_t*>(rdft_result),
                               stft_shape,
                               stft_transp_out_shape,
                               sizeof(float));
}
```

Reply:
Needed to slightly adjust the proposed change, but I got the idea.
Implemented the usage ScratchPadMem for intermediate results if transpose is needed, and eventually I moved
the code of `stft_impl` func to `STFT::exececute` as the `getScratchPadMem` is `protected`.
I've decided to skip related changes in prepareParams at this point, as it could be pre-defined only for static shape case.
Please take a look at the latest changes, especially commit https://github.com/openvinotoolkit/openvino/pull/26967/commits/5a8c341c310bb03df6017b1258ed709ab7649823.
Thanks!